gensim/matutils.py

```diff
@@ -382,6 +383,112 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    Vec is considered to be in bag of words format if it is 2-tuple format.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    try:
+        id_, val_ = vec[0] # checking first value to see if it is in bag of words format by unpacking
+        id_, val_ = int(id_), float(val_)
+    except IndexError:
+        return True # this is to handle the empty input case
+    except Exception:
+        return False
+    return True
+
+
+def kullback_leibler(vec1, vec2, num_features=None):
+    """
+    A similarity metric between two probability distributions.
+    Returns a similarity in range <0,1> where values closer to 0 mean a higher similarity.
+    Uses the scipy.stats.entropy method to identify kullback_leibler convergence value.
+    If the distribution draws from a certain number of docs, that value must be passed.
+    """
+    if scipy.sparse.issparse(vec1):
+        vec1 = numpy.asarray(vec1.todense())
```

Comment:
Isn't simply `sparse.toarray` better? (`todense` returns a matrix, not sure why we'd want that).